### PR TITLE
Gradle script to gzip all bundles js files and spring configuration t…

### PR DESCRIPTION
**Pull request by bgsavage**
…o serve them when found.

### CODE REVIEW

File: webapp/build.gradle [hunk 1]
```diff
@@ -62,22 +62,34 @@ test.finalizedBy(jacocoTestReport)
...
 }
 
+task gzipAllJs(dependsOn: buildFe) {
+    doLast {
+        FileTree tree = fileTree(dir: "${project.projectDir}/src/main/resources/static/", include: "**/*.bundle.js")
+        tree.each { File file ->
+            String fileName = file.getName()
+            println "$fileName => ${fileName}.gz"
+            ant.gzip(src: file, destfile: file.toString() + ".gz")
+        }
```
Comment (bgsavage) [inline webapp/build.gradle]: I spent a long time trying to get a 3rd party plugin (gradle-js) gzip task to work with multiple files only later to find out that there's an ant.gzip method I could use instead...

File: webapp/src/main/resources/application.yml [hunk 2]
```diff
@@ -31,6 +31,11 @@ server:
     max-threads: 20
 
 spring:
+  resources:
+    chain:
+      enabled: true
+      gzipped: true
```
Comment (bgsavage) [inline webapp/src/main/resources/application.yml]: Configuration in which spring will look for a `.gz` variant of any resource requested and will serve that instead if it finds one.

File: webapp/build.gradle (hunk 1 above)
Comment (rcordeau) [reply webapp/build.gradle]: So you want one .gz per bundle.js file?

File: webapp/src/main/resources/application.yml (hunk 2 above)
Comment (jtreuting) [reply webapp/src/main/resources/application.yml]: does this flag also tell Spring to do the gzip on the fly?  i'm guessing no, which i think is good for the first step

File: webapp/build.gradle (hunk 1 above)
Comment (jtreuting) [reply webapp/build.gradle]: yeah, if there is a somefile.js.gz file on the server, spring can return it for browsers that make a request for somefile.js and have the Accept-Encoding header of gzip.  brandon said it made the vendor bundle go down from 1.4 MG to 400K so a lot faster response to load the initial page for users

Comment (bgsavage) [reply webapp/build.gradle]: Yeah I'm thinking we still want the separate bundles so they can be cached individually and grabbed in parallel.

File: webapp/src/main/resources/application.yml (hunk 2 above)
Comment (bgsavage) [reply webapp/src/main/resources/application.yml]: Nope, server.compression.enabled tells spring to do the gzip on the fly for a list of specified MIME types, which is not in this PR.

File: webapp/build.gradle (hunk 1 above)
Comment (rcordeau) [reply webapp/build.gradle]: Sounds good to me. Faster responses to loading is great.

**Pull request merged**